### ISSUE TITLE
can asp(n:m sparsity) works along with hugging face?

### ISSUE THREAD

**Issue by LeiWang1999** [NVIDIA/apex#1497]
I'm trying to prune a n:m sparse bert-large based on hugging face, so can we prune the network like we did in pytorch as well?  

And could anybody please tell me is there an open source project to replicate the content in this blog post
https://developer.nvidia.com/blog/accelerating-inference-with-sparsity-using-ampere-and-tensorrt/

**Comment by LeiWang1999** [comment]
yes, it works now.

**Comment by cschumacher-ias** [comment]
Hey @LeiWang1999, I've been wrestling with a similar issue with a roberta model from HF. I was able to make sparsity with this model by tweaking a few lines:
- swapping the symbolic_trace import on [893](https://github.com/NVIDIA/apex/blob/master/apex/contrib/sparsity/permutation_lib.py#L893) to be from transformers.utils.fx instead of torch.fx.
- tweaking the condition on [665](https://github.com/NVIDIA/apex/blob/master/apex/contrib/sparsity/permutation_lib.py#L665) to ensure child_name isn't 'output' or 'None'. It appears there is an [open PR](https://github.com/NVIDIA/apex/pull/1441) that may fix this issue in the future.

While I was able to enable sparse masking and restore the performance with only a couple of epochs of training, I haven't been able to get sparsity-based optimizations activated when utilizing the TRT backend with amphere gpus, even when using `--sparsity=force' with trtexec. After some digging, it appears that transformer architectures default to utilization of the myelin kernel with TRT (reference [here](https://github.com/NVIDIA/TensorRT/issues/2271#issuecomment-1229359118)). 

Would you mind sharing the solution you came up with and confirm if you were able to get sparsity-optimizations for bert active within TRT?

**Comment by LeiWang1999** [comment]
@cschumacher-ias sorry for the late reply, for our works, wo do not enbale sparse tensorcore, but actually we benchmark a 50% sparsity model with trtexec but we only got a dense performance
